### ISSUE TITLE
Better starter kit

### ISSUE THREAD

**Issue by mariusandra** [PostHog/plugin-server#17]
Currently the [helloworldplugin](https://github.com/PostHog/helloworldplugin) is our "blank plugin starter kit".

This should be extended with:
- TypeScript support (copied from the maxmind plugin)
- Tests (#16)

**Comment by Twixes** [comment]
Done, since we've now got https://github.com/PostHog/posthog-plugin-starter-kit & https://github.com/PostHog/posthog-plugin-advanced-kit.

**Comment by mariusandra** [comment]
Well, not yet done since those starter kits don't work with the current version of posthog nor plugin server

**Comment by Twixes** [comment]
Ah right, though I think they are only missing `plugin.json`

**Comment by mariusandra** [comment]
Well, I could come up with a list of (subjective?) things to improve... but these are not something to spend time on now.

- The entire `plugin.json` situation.
- The naming is still confusing to me. What's "advanced"? For me the only legit difference is JS vs TS, so the name should just reflect that.
- I'd still prefer `yarn.lock` over `package-json.lock`, just to be consistent with every other posthog repository. If you are using npm and run `npm install`, nothing bad will happen. NPM 7 even supports `yarn.lock` out of the box (even though it'll still make a `package-lock.json`).
- There are probably too many files in the simple starter kit. `.editorconfig` and `.gitattributes` can probably go. `jest.config.js` can be moved into `package.json`.
- The maxmind plugin required rollup to mash all imported ts files into one. As it is now, you can't move your shared functions into another folder and just import them. That begs the question: what's the use of a `src` folder if it can only contain one file?
- The TS starter doesn't contain guides to help you with typing your `global`, `config` and `attachments` like in the [maxmind plugin](https://github.com/PostHog/posthog-maxmind-plugin/blob/08519d2395803b6da0565f44c2cb8b7637901e2b/src/index.ts#L4).
- I still think the TS plugin structure in the starter [as it is now](https://github.com/PostHog/posthog-plugin-advanced-kit/blob/a954ce9cf06274ea6525dc6d7102948072ffb763/src/index.ts#L18), while providing better typing support, is less friendly to the developers. Having just a bunch of predefined functions and adding `: Meta` to the arguments is not that much work and makes everything look so much cleaner and simpler. But I understand we will not reach consensus here.

Again, let's not work on this now. There are bigger fish to fry.

**Comment by Twixes** [comment]
Generally agree (besides `plugin.json`, but let's solve this by going with vox populi). As for naming, not set on it all (though just `typescript` it's a bit limiting, that's just a part of it).
And definitely +1 on using Rollup.
That tiny additional bit of typing makes too nice of a difference to just give up though!

**Comment by mariusandra** [comment]
With plugins becoming more and more mainstream, people tend to take whatever random starting point they can find and code a plugin on top of this. 

To remove confusion and help adoption of plugins, I propose it's finally time to clean up this starter kit situation (together with world class docs)... and consolidate on just one and _simplified_ `posthog-plugin-starter-kit` that we promote everywhere.

My wishes for this kit:
- As friendly and approachable as possible. 
- Not much to delete when starting the project
- TypeScript and thus no compilation step with rollup or anything
- Yarn and `yarn create posthog-plugin`
- Remove all unnecessary yet nice to have files like `.editorconfig`.
- Keep all the code under `src`, keep simple jest tests under `tests`. No `dist` needed.
- Investigate the option of even moving `plugin.json` under `src`, and adding `src/index.ts` as another search path for the code in an archive.

**Comment by neilkakkar** [comment]
So, I just tried writing a new plugin, and this template still _feels_ annoying.

I followed the instructions in the tutorial ( https://github.com/PostHog/posthog.com/pull/1467 ), and once I used the template, I ended up deleting lots of files that felt useless ( https://github.com/neilkakkar/posthog-github-followers-plugin/commits/main )

This included: the test file, package.json, yarn.lock. I didn't really need these to finish my plugin, since I'm testing it using "Install from URL" (everything is happening in the browser).

I think the crux here is that: **the template shouldn't be a plugin in itself, but just a starting point**.

This implies getting rid of everything extra in the plugin - tests, package.json, lockfiles, etc. 

I think we should split things out into: 

(1) Exemplary plugins to serve as reference. This would include a list of plugins and their specialty, showing how tests and package.json work.
(2) Template - the starter kit, which quickly lets you get up and running with a plugin.

I'm planning to include a link to (1) from (2), so everything is easily available, and then all our technique improvements / upgrades actually go into existing plugins, instead of the template. Something like:

![image](https://user-images.githubusercontent.com/7115141/122211411-98e17880-ce9e-11eb-9032-1da4d943b3f0.png)
